ga4gh/exceptions.py

```diff
@@ -0,0 +1,179 @@
+"""
+Exceptions for the GA4GH server. Each exception that can occur in the server
+is given a unique error code that is derived from its name.
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import zlib
+
+import ga4gh.protocol as protocol
+
+
+def convertException(exception):
+    """
+    Converts the specified exception that is not a subclass of the
+    ServerException into a ServerError so that it can be correctly
+    serialised and communicated back to the client.
+    """
+    serverException = ServerError()
+    serverException.message = str(exception)
+    return serverException
+
+
+class ServerException(Exception):
```

Comment:
maybe rename this class to make the difference with ServerError more obvious?  Like BaseServerException or something